patina_dxe_core/src/lib.rs

```diff
@@ -541,10 +541,15 @@ impl<P: PlatformInfo> Core<P> {
     /// Clippy gets unhappy if we call get_c_hob_list_size directly, because it gets confused, thinking
     /// get_c_hob_list_size is not marked unsafe, but it is
     fn get_hob_list_len(hob_list: *const c_void) -> usize {
+        // SAFETY: hob_list is checked for NULL before the call to get_hob_list_len
```

Comment:
While here, you could try removing this wrapper function and call the unsafe func directly and see if clippy is still confused